main.go

```diff
@@ -92,10 +95,114 @@ func startAccepting(maxClients int64, connType, connHost, connPort string, inter
 				LocalPort:  connPort,
 			}
 			clients <- c
+			go reportIPToAbuseIPDB(remoteIpAddr, records, abuseipdbeEnabled, abuseIpdbApiKey)
 		}
 	}()
 }
 
+func reportIPToAbuseIPDB(ip string, records chan<- metrics.RecordEntry, abuseipdbeEnabled bool, abuseIpdbApiKey string) {
+	if !abuseipdbeEnabled {
+		return
+	}
+	if isCached(ip) {
+		glog.V(1).Infof("IP is already cached, skipping report")
+		records <- metrics.RecordEntry{
+			RecordType: metrics.RecordEntryTypeReport,
+			IpAddr:     ip,
+			Message:    "IP is already cached, skipping report",
+		}
+		return
+	}
+
+	var apiKey string
+	abuseIpdbApiKeyFile := "ABUSE_IPDB_API_KEY_FILE"
+	if abuseIpdbApiKey != "" {
+		apiKey = abuseIpdbApiKey
+	} else if abuseIpdbApiKeyFile != "" {
+		key, err := os.ReadFile(abuseIpdbApiKeyFile)
```

Comment:
This is not how it should work, I believe. `ABUSE_IPDB_API_KEY_FILE` should be an environment variable, pointing to a path with a file including the key.